### DISCUSSION TITLE
How to create custom `StraightEdge` ?

### DISCUSSION THREAD

**Discussion by nowiko** [xyflow/xyflow#2109] [Q&A] [answered]
Hi, sorry if this is an obvious question, but it's not quite clear to me how to create a custom `StraightEdge`.  Accordingly to [example](https://reactflow.dev/docs/examples/edges/) , if I am using `getSmoothStepPath` instead of `getBezierPath`, it renders smooth edge correctly.  There is no additional function like `getStraightPath`, how I can achieve the desired result?

**Comment by bcakmakoglu** [comment, marked as answer]
Well you just use a straight line which doesn't require you to use `getBezierPath`  or the other utilities 😅 
Here's what the implementation of the straight edge looks like.
`EdgeProps` are available in your custom edge as well.
```tsx
memo(
  ({
    sourceX,
    sourceY,
    targetX,
    targetY,
    label,
    labelStyle,
    labelShowBg,
    labelBgStyle,
    labelBgPadding,
    labelBgBorderRadius,
    style,
    markerEnd,
    markerStart,
  }: EdgeProps) => {
    const yOffset = Math.abs(targetY - sourceY) / 2;
    const centerY = targetY < sourceY ? targetY + yOffset : targetY - yOffset;

    const xOffset = Math.abs(targetX - sourceX) / 2;
    const centerX = targetX < sourceX ? targetX + xOffset : targetX - xOffset;
    const path = `M ${sourceX},${sourceY}L ${targetX},${targetY}`;

...
})
```

Basically copy paste that into your custom edge and you're done, you got yourself a straight edge 😄

**Reply by nowiko** [reply]
Thanks for the answer, I will try to apply this solution

**Reply by nowiko** [reply]
@bcakmakoglu yep, this code works, thanks for help.
